systemtest/containers.go

```diff
@@ -234,3 +240,263 @@ func (c *ElasticsearchContainer) Start() error {
 func (c *ElasticsearchContainer) Close() error {
 	return c.container.Terminate(context.Background())
 }
+
+// NewUnstartedElasticAgentContainer returns a new ElasticAgentContainer.
+func NewUnstartedElasticAgentContainer() (*ElasticAgentContainer, error) {
+	// Create a testcontainer.ContainerRequest to run Elastic Agent.
+	// We pull some configuration from the Kibana docker-compose service,
+	// such as the Docker network to use.
+
+	docker, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return nil, err
+	}
+	defer docker.Close()
+
+	kibanaContainer, err := stackContainerInfo(context.Background(), docker, "kibana")
+	if err != nil {
+		return nil, err
+	}
+	kibanaContainerDetails, err := docker.ContainerInspect(context.Background(), kibanaContainer.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	var kibanaIPAddress string
+	var networks []string
+	for network, settings := range kibanaContainerDetails.NetworkSettings.Networks {
+		networks = append(networks, network)
+		if kibanaIPAddress == "" && settings.IPAddress != "" {
+			kibanaIPAddress = settings.IPAddress
+		}
+	}
+	kibanaURL := &url.URL{
+		Scheme: "http",
+		Host:   net.JoinHostPort(kibanaIPAddress, apmservertest.DefaultKibanaPort()),
+	}
+
+	// Use the same stack version as used for Kibana.
+	agentImageVersion := kibanaContainer.Image[strings.LastIndex(kibanaContainer.Image, ":")+1:]
+	agentImage := "docker.elastic.co/beats/elastic-agent:" + agentImageVersion
+	if err := pullDockerImage(context.Background(), docker, agentImage); err != nil {
+		return nil, err
+	}
+	agentImageDetails, _, err := docker.ImageInspectWithRaw(context.Background(), agentImage)
+	if err != nil {
+		return nil, err
+	}
+	agentVCSRef := agentImageDetails.Config.Labels["org.label-schema.vcs-ref"]
+	agentDataHashDir := path.Join("/usr/share/elastic-agent/data", "elastic-agent-"+agentVCSRef[:6])
+	agentDownloadsDir := path.Join(agentDataHashDir, "downloads")
+	agentInstallDir := path.Join(agentDataHashDir, "install")
+
+	// Build elastic-agent to replace the binary inside the container.
+	//
+	// This enables us to inject a locally built, unsigned, apm-server artifact.
+	//
+	// TODO(axw) once apm-server is bundled we can stop building a custom
```

Comment:
long time has passed since you did this, but we should not need this now

Reply:
Nice! Evidently I forgot to subscribe to the bundling issue. I've removed this code now.